mne/preprocessing/ecg.py

```diff
@@ -376,14 +393,35 @@ def _make_ecg(inst, start, stop, reject_by_annotation=False, verbose=None):
                 .format({'mag': 'Magnetometers',
                          'grad': 'Gradiometers'}[ch]))
     picks = pick_types(inst.info, meg=ch, eeg=False, ref_meg=False)
+
+    # Handle start/stop
+    begin_param_name = 'tmin'
+    if isinstance(start, int_like):
+        if isinstance(inst, BaseRaw):
+            # Raw has start param, can just use int
+            start = 0 if start is None else start
+            begin_param_name = 'start'
+        # for Epochs/Evoked, convert time idx to time in s
+        start = 99999.
```

Comment:
I would rather just not support sample numbers for epochs/evoked. My thinking is that we really only want to / need to support sample numbers for raw data. For epochs and evoked we should always work in time units. Would this work?

Reply:
yes definitely ... this is just a placeholder: 999999. is a nonsensical value: What I need at that place is a `start = inst.index_to_time(start)` method, see: https://github.com/mne-tools/mne-python/pull/9556#discussion_r670472858

Reply:
> What I need at that place is

I saw that, I'm saying let's not do it at all :) I don't think we really want to start supporting index-based time-axis access for Epochs and Evoked if we can avoid it

Reply:
ahh okay, I'll see. it'll probably be less pretty - but let's see.

Reply:
I think you can check for int-like-ness; if not instance of raw, raise, otherwise convert. Should be just a few lines...

Reply:
mmh why should we raise? 

- if None --> take start/end of signal
- if int --> integer index into signal
- if float --> time (s) index into signal

for Raw, all of this is easily suported via start/stop and tmin/tmax

For Epochs/Evoked everything works as well except INT (no start/stop params available there) ... but I could just directly index into the data, right? Why not?

What I meant with not pretty is that:
1. it's "custom indexing code" and
2. need to somehow handle the case where a combination of float and int is passed.

Or are you suggesting that for Epochs/Evoked we only accept floats and None, and raise for ints?

Reply:
> Or are you suggesting that for Epochs/Evoked we only accept floats and None, and raise for ints?

Yes exactly

> For Epochs/Evoked everything works as well except INT (no start/stop params available there) ... but I could just directly index into the data, right? Why not?

Every use case we support increases our maintenance burden, and also cognitive load for users. I doubt any user really wants to index these objects using ints and if they do, their use case is rare and they can round/etc. as they see fit. I kind of regret having sample-level options even for raw, but there are at least some use cases there and we have to for backward compat.

Reply:
re-reading,

> Or are you suggesting that for Epochs/Evoked we only accept floats and None, and raise for ints?

you probably are :laughing: 

I'll push a commit

Reply:
okay, done --> Currently with CI Skip, because I haven't implemented tmin/tmax for Epochs yet --> will do after the "Units" for Epochs are in